lms/djangoapps/course_goals/models.py

```diff
@@ -46,3 +46,20 @@ def __str__(self):
             goal=self.days_per_week,
             course=self.course_key,
         )
+
+
+class UserActivity(models.Model):
```

Comment:
if we remove the activity type field and if we want to include the mode in course goals, i'd lobby we call it something like UserGoalsActivity or something

the name UserActivity implies this is a pretty generic table that could be used for anything (which I think was a possible consideration), but if we don't store activity_type, we can only store visits to the right subset of pages in this table in order to not break goals - which would not be immediately obvious to me if I just saw a model called user activity

Reply:
I understand what you're saying, however I think because both the djangoapp and the table in the database will have course_goals in their name, it will be clear to other users that this table is not generic and is specifically for goals (as opposed to being in say the student djangoapp). Whether it's page views as implemented or something else in the future if we add in additional information, the table is still tracking on behalf of goals.